datastr/dict.go

```diff
@@ -17,16 +18,23 @@ const (
 type Dict struct {
 	hashTables [2]*HashTable
 	rehashidx  int
+	rwmux sync.RWMutex
+	randomBytes [16]byte
+	once sync.Once
+	key0 uint64
+	key1 uint64
+	logger *darelog.LOG
 }
 
 // NewDict returns a new instance of Dict.
 //
 // The function does not take any parameters.
 // It returns a pointer to Dict.
-func NewDict() *Dict {
+func NewDict(logger *darelog.LOG) *Dict {
```

Comment:
IMHO currently I don't want to log anything because I'm focusing on performances and logging can increase cpu and memory usage. Furthermore, darelog is just a WIP and not ready to use.

Reply:
i can't code without proper logging
cpu and memory usage is low as f.. *g
i love it!